src/tree/ResolvableTreeItemBase.ts

```diff
@@ -28,9 +29,17 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
         return this.resolveResult?.description;
     }
 
+    public get collapsibleState(): TreeItemCollapsibleState {
+        // TODO: verify this is correct
+        return this.resolveResult?.collapsibleState ?? !!this.resolveResult?.loadMoreChildrenImpl ? TreeItemCollapsibleState.Collapsed : TreeItemCollapsibleState.None;
```

Comment:
The goal is for the initial state of the node to be `collapsibleState === TreeItemCollapsibleState.None`, i.e. it will have no expand arrow. Once it resolves, if the resolve attached a `loadMoreChildrenImpl` method, then it gets its arrow. I think the UX of having a spinner for a short amount of time before the arrow appears looks fine.

Reply:
(That said I'm not sure this is even the correct way to accomplish that goal, which is why the comment is there)

Reply:
VS Code only uses the collapsible state of a tree item when it's created, and then never uses the value again, even after refreshing / firing the data changed event. https://github.com/microsoft/vscode/issues/127035#issuecomment-877050461

Reply:
Marking this as resolved since it seems to be really using that value.